src/ChannelServer/World/TasksPoolManager.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Melia.Channel.World
+{
+
+	public class CbFunctionData
+	{
+		public TasksPoolManager.TimerCallbackFunction cbFunction;
+		public Object obj;
+	}
+
+	public class TasksPoolManager
```

Comment:
What's the specific purpose for creating your own task manager, is something wrong about `System.Threading.Tasks.Task`?

Reply:
I dunno how to use that

Reply:
Instead of that entire custom class you would for example do this:
```csharp
Task.Delay(this.Data.respawnTime * 1000).ContinueWith(this.DoSpawn);
```
https://msdn.microsoft.com/en-us/library/system.threading.tasks.task%28v=vs.110%29.aspx?f=255&MSPPError=-2147217396